x/concentrated-liquidity/grpc_query.go

```diff
@@ -211,6 +211,39 @@ func (q Querier) TotalLiquidityForRange(goCtx context.Context, req *clquery.Quer
 	return &clquery.QueryTotalLiquidityForRangeResponse{Liquidity: liquidity}, nil
 }
 
+// LiquidityNetInDirection returns an array of LiquidityDepthWithRange, which contains the range(lower tick and upper tick) and the liquidity amount in the range.
+func (q Querier) LiquidityNetInDirection(goCtx context.Context, req *clquery.QueryLiquidityNetInDirectionRequest) (*clquery.QueryLiquidityNetInDirectionResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// convert values from pointers
+	var boundTick sdk.Int
+	if req.BoundTick == nil {
+		boundTick = sdk.Int{}
+	} else {
+		boundTick = *req.BoundTick
+	}
+
+	liquidityDepths, err := q.Keeper.GetLiquidityNetInDirection(
+		ctx,
+		req.PoolId,
+		req.TokenIn,
+		boundTick,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	pool, err := q.Keeper.getPoolById(ctx, req.PoolId)
+	if err != nil {
+		return nil, err
+	}
+
+	return &clquery.QueryLiquidityNetInDirectionResponse{LiquidityDepths: liquidityDepths, CurrentLiquidity: pool.GetLiquidity()}, nil
```

Comment:
It seems to me that we include "current_tick" and it's liquidity net in the `liquidityDepths`. However, current tick's `liquidity_net` is irrelevant for FE swap estimate.

In terms of the current tick, we only need to know a) the current tick index b) current tick liquidity (which we already return)

So I suggest only including liquidity net amounts starting from the "next tick" from current in the swap direction

Instead, we can have a separate field called `CurrentTick` similar to `Current`Liquidity`

Let me know what you think

Reply:
CC: @jonator for awareness on this suggestion